### PR TITLE
fix delint conflicts

### PR DESCRIPTION
**Pull request by AshesITR**
Unfortunately, some newly enabled linters had hits in `tests`, so we aren't lint-free on master atm.

### CODE REVIEW

File: .lintr [hunk 1]
```diff
@@ -1,6 +1,6 @@
 linters: with_defaults( # The following TODOs are part of an effort to have {lintr} lint-free (#584)
    line_length_linter = line_length_linter(120),
-   cyclocomp_linter = cyclocomp_linter(29), # TODO reduce to 15
+   cyclocomp_linter = cyclocomp_linter(30), # TODO reduce to 15
```
Comment (AshesITR) [inline .lintr:3]: #673 added 1 to the cyclocomp of `lint()`.
To keep this separate from removing ~ 200 lints from tests/, I temporarily bump the limit here.

**Review by MichaelChirico — DISMISSED** [review]

File: tests/testthat/test-error.R [hunk 2]
```diff
@@ -60,11 +60,6 @@ test_that("returns the correct linting", {
   expected_message <- tryCatch(parse(text = "\\"), error = get_base_message)
   expect_lint("\\", rex(expected_message))
 
-  # also try when LANGUAGE initially unset
```
Comment (AshesITR) [inline tests/testthat/test-error.R:63]: This causes non-english locales to fail the test.

**Comment by AshesITR** [comment]
@MichaelChirico can you re-approve? I had a merge conflict and test failure on non-english locale due to `Sys.unsetenv()` in `test-error.R`.

**Review by MichaelChirico — APPROVED** [review]

**Comment by MichaelChirico** [comment]
That `Sys.unsetenv` was for coverage -- do we know what's the issue exactly?

**Comment by AshesITR** [comment]
What issue do you mean?
On non-english locales, Sys.unsetenv will create essentially unpredictable messages and the following tests check for these.
I think what would work would be something like

```
Sys.unsetenv("LANGUAGE")
get non-english expected error message
test in non-english locale
set_lang("en")
```

The old code was missing the `set_lang("en")` and the update of the expected error message.

**Comment by MichaelChirico** [comment]
OK, we'll add that back in the future

**Pull request merged**